lib/generators/manageiq/provider/provider_generator.rb

```diff
@@ -25,23 +44,37 @@ def create_provider_files
 
     alias provider_name file_name
 
+    def validate_manager_type!
+      return unless manager_type.nil?
+
+      raise "Invalid manager_type: #{options[:manager_type]}\nMust be one of #{self.class.manager_types.keys.join(", ")}"
```

Comment:
I think you forgot the conditional here?

Reply:
It returns if the manager_type is valid, the line was getting a little long to be sticking an `unless manager_type` at the end

Reply:
But I don't see an actual check against the list.  I only see the nil check but if someone passes `--manager-type=skdjhfd` then I think this is validated true.

Reply:
`#manager_type` does `self.class.manager_types[options[:manager_type]]` so `manager_type.nil?` means that it isn't in the list.

```
$ rails generate manageiq:provider ManageIQ::Providers::SomeVendor --manager-type=asfasldfsaf --dummy
/home/grare/adam/src/manageiq/manageiq/lib/generators/manageiq/provider/provider_generator.rb:49:in `validate_manager_type!': Invalid manager_type: asfasldfsaf (RuntimeError)
Must be one of automation, cloud, configuration, container, infra, monitoring, network, physical, provisioning, storage
```

Reply:
Oh ok, I didn't realize manager_type was a method with side-effect...I thought it was literally the value passed in by the caller. :+1: